### PR TITLE
[OTEL-2101] Start ddflare extension even if there is no DD exporter config

### PR DESCRIPTION
**Pull request by songy23**
### What does this PR do?

Start ddflare extension even if there is no DD exporter config

### Describe how to test/QA your changes

Will be covered in the E2E test

### CODE REVIEW

File: cmd/otel-agent/subcommands/run/command.go [hunk 1]
```diff
@@ -96,10 +96,16 @@ func runOTelAgentCommand(ctx context.Context, params *subcommands.GlobalParams,
...
 	}
+	uris := append(params.ConfPaths, params.Sets...)
 	if err == agentConfig.ErrNoDDExporter {
 		return fxutil.Run(
-			fx.Provide(func() []string {
-				return append(params.ConfPaths, params.Sets...)
+			fx.Supply(uris),
+			fx.Provide(func() optional.Option[coreconfig.Component] {
+				return optional.NewNoneOption[coreconfig.Component]()
+			}),
```
Comment (dustmop) [inline cmd/otel-agent/subcommands/run/command.go:105]: You could replace this with the shorter version:
```suggestion
			fx.Supply(optional.NewNone[coreconfig.Component]),
```

File: cmd/otel-agent/subcommands/run/command.go [hunk 2]
```diff
@@ -128,13 +134,14 @@ func runOTelAgentCommand(ctx context.Context, params *subcommands.GlobalParams,
 			pkgconfigenv.DetectFeatures(acfg)
 			return acfg, nil
 		}),
+		fx.Provide(func(corecfg coreconfig.Component) optional.Option[coreconfig.Component] {
+			return optional.NewOption[coreconfig.Component](corecfg)
+		}),
```
Comment (dustmop) [inline cmd/otel-agent/subcommands/run/command.go:139]:
```suggestion
		fx.ProvideOptional[coreconfig.Component](),
```

**Review by dustmop — COMMENTED** [review]
LGTM, some suggestions

**Review by jackgopack4 — APPROVED** [review]
looks good to me

File: cmd/otel-agent/subcommands/run/command.go [hunk 3]
```diff
@@ -96,10 +96,14 @@ func runOTelAgentCommand(ctx context.Context, params *subcommands.GlobalParams,
 	if err != nil && err != agentConfig.ErrNoDDExporter {
 		return err
 	}
+	uris := append(params.ConfPaths, params.Sets...)
 	if err == agentConfig.ErrNoDDExporter {
 		return fxutil.Run(
-			fx.Provide(func() []string {
-				return append(params.ConfPaths, params.Sets...)
+			fx.Supply(uris),
+			fx.Supply(optional.NewNone[coreconfig.Component]),
```
Comment (songy23) [inline cmd/otel-agent/subcommands/run/command.go:103]:
```suggestion
			fx.Supply(optional.NewNoneOption[coreconfig.Component]),
```

File: cmd/otel-agent/subcommands/run/command.go [hunk 4]
```diff
@@ -128,13 +132,12 @@ func runOTelAgentCommand(ctx context.Context, params *subcommands.GlobalParams,
 			pkgconfigenv.DetectFeatures(acfg)
 			return acfg, nil
 		}),
+		fx.ProvideOptional[coreconfig.Component](),
```
Comment (songy23) [inline cmd/otel-agent/subcommands/run/command.go:135]:
```suggestion
		fxutil.ProvideOptional[coreconfig.Component](),
```

**Review by dineshg13 — APPROVED** [review]

**Pull request merged**